infer/modules/vc/pipeline.py

```diff
@@ -362,7 +362,7 @@ def pipeline(
             )
             pitch = pitch[:p_len]
             pitchf = pitchf[:p_len]
-            if self.device == "mps":
+            if self.device == "mps" or (hasattr(torch, "xpu") and torch.xpu.is_available()):
```

Comment:
Same, one time `(hasattr(torch, "xpu") and torch.xpu.is_available())` calculation, read cached result everywhere.

Reply:
Done.